packages/plugin-compiler/src/main/kotlin/io/realm/kotlin/compiler/RealmModelDefaultMethodGeneration.kt

```diff
@@ -0,0 +1,123 @@
+package io.realm.kotlin.compiler
+
+import org.jetbrains.kotlin.backend.common.extensions.IrPluginContext
+import org.jetbrains.kotlin.ir.builders.irGet
+import org.jetbrains.kotlin.ir.builders.irGetObject
+import org.jetbrains.kotlin.ir.builders.irReturn
+import org.jetbrains.kotlin.ir.declarations.IrClass
+import org.jetbrains.kotlin.ir.declarations.IrProperty
+import org.jetbrains.kotlin.ir.declarations.IrSimpleFunction
+import org.jetbrains.kotlin.ir.expressions.impl.IrCallImpl
+import org.jetbrains.kotlin.ir.types.IrType
+import org.jetbrains.kotlin.ir.util.functions
+import org.jetbrains.kotlin.name.Name
+
+/**
+ * Class responsible for adding Realm specific logic to the default object methods like:
+ * - toString()
+ * - hashCode()
+ * - equals()
+ *
+ * WARNING: The current logic in here does not work well with incremental compilation. The reason
```

Comment:
It could also be something else, but at least I had issues in the tests that was fixed by a clean build. Note, the problem was bigger when I inlined the functions, now we can change the implementation in RealmObjectHelper and have it working, so the chance of this happening are pretty low.